pkg/integration/replication/synchronous_replication_test.go

```diff
@@ -102,59 +116,49 @@ func (suite *SyncTestSuite) TestMasterRestart() {
 	})
 }
 
+type SyncTestSuitePrecommitStateSync struct {
+	baseReplicationTestSuite
+}
+
+func TestSyncTestSuitePrecommitStateSync(t *testing.T) {
+	suite.Run(t, &SyncTestSuitePrecommitStateSync{})
+}
+
+// this function executes before the test suite begins execution
+func (suite *SyncTestSuitePrecommitStateSync) SetupTest() {
+	suite.baseReplicationTestSuite.SetupTest()
+	suite.SetupCluster(2, 2, 0)
+}
+
 // TestPrecommitStateSync checks if the precommit state at master
 // and its followers are in sync during synchronous replication
-func (suite *SyncTestSuite) TestPrecommitStateSync() {
-	var (
-		masterState *schema.ImmutableState
-		err         error
-		startCh     = make(chan bool)
-	)
-
+func (suite *SyncTestSuitePrecommitStateSync) TestPrecommitStateSync() {
 	ctx, client, cleanup := suite.ClientForMaster()
 	defer cleanup()
 
-	// Create goroutines for client waiting to query the state
-	// of the followers. This is initialised before to avoid
-	// spending time initialising the follower client for faster
-	// state access
-	var wg sync.WaitGroup
-	for i := 0; i < suite.GetFollowersCount(); i++ {
```

Comment:
This should be restored - it's about reading replicas state as soon as possible right after the commit is confirmed.